File: frontend/src/types/recommendation.ts
export type DimensionContribution = {
  dim: string;
  name: string;
  contribution: number;
};

export type Recommendation = {
  movie_id: number;
  title: string;
  year: number;
  genres: string[];
  match_score: number;
  similarity_dims: number[];
  why_recommended: string;
};

export type RecommendationExplanation = {
  cosine_similarity: number;
  top_dims: DimensionContribution[];
  neighbor_users: number[];
};

File: backend/app/schemas/recommendation.py
from pydantic import BaseModel
from typing import List

class DimensionContribution(BaseModel):
    dim: str
    name: str
    contribution: float

class RecommendationResponse(BaseModel):
    movie_id: int
    title: str
    year: int
    genres: List[str]
    match_score: float
    similarity_dims: List[float]
    why_recommended: str

class ExplanationResponse(BaseModel):
    cosine_similarity: float
    top_dims: List[DimensionContribution]
    neighbor_users: List[int]
